modules/howtos/examples/transactions-example.php

```diff
@@ -0,0 +1,489 @@
+          $ctx->remove($monsterDoc);
+
+          // The player earns experience for killing the monster
+          $experienceForKillingMonster = $monsterContent["experienceWhenKilled"];
+          $playerExperience = $playerContent["experience"];
+          $playerNewExperience = $playerExperience + $experienceForKillingMonster;
+          $playerNewLevel =
+            calculateLevelForExperience($playerNewExperience);
+
+          $playerContent['experience'] = $playerNewExperience;
+          $playerContent['level'] = $playerNewLevel;
+
+          $ctx->replace($playerDoc, $playerContent);
+      }
+    });
+  }
+  catch (\Couchbase\Exception\TransactionOperationFailedException $e) {
+    echo "Transaction did not reach commit point\n";
+    // The operation failed. Both the monster and the player will be untouched.
+    //
+    // Situations that can cause this would include either the monster
+    // or player not existing (as get is used), or a persistent
+    // failure to be able to commit the transaction, for example on
+    // prolonged node failure.
+  }
+  catch (\Couchbase\Exception\TransactionException $e) { 
+    // TODO check is this equivalent to TransactionCommitAmbiguousError as per Node examples?
+    // Indicates the state of a transaction ended as ambiguous and may or
+    // may not have committed successfully.
+    //
+    // Situations that may cause this would include a network or node failure
+    // after the transactions operations completed and committed, but before the
+    // commit result was returned to the client
+    echo "Transaction possibly committed\n";
+  }
+}
+// end::full[]
+
+function rollbackExample($cluster, $collection) {
+  $costOfItem = 10;
+  
+  // tag::rollback[]
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) use ($collection, $costOfItem) {
+      $customer = $ctx->get($collection, "customer-name");
+      if ($customer->content()["balance"] < $costOfItem) {
+        throw new \Error("Transaction failed, customer does not have enough funds.");
+      }
+      // else continue transaction
+  });
+  // end::rollback[]
+}
+
+class InsufficientBalanceError extends Error {}
+
+function rollbackCause($cluster, $collection) {
+  $costOfItem = 10;
+
+  // tag::rollback-cause[]
+  try {
+    $cluster->transactions()->run(
+      function (TransactionAttemptContext $ctx) use ($collection, $costOfItem) {
+        $customer = $ctx->get($collection, "customer-name");
+
+        if ($customer->content()["balance"] < $costOfItem) {
+          throw new \InsufficientBalanceError("Transaction failed, customer does not have enough funds.");
+        }
+        // else continue transaction
+    });
+  }
+  catch (\Couchbase\Exception\TransactionOperationFailedException $e) {
+    // This exception can only be thrown at the commit point, after the
+    // BalanceInsufficient logic has been passed
+  }
+  catch (InsufficientBalanceError $e) {
```

Comment:
I've already raised with Sergey, but just to reiterate, it should be a TransactionFailedException raised, with a cause of InsufficientBalanceError.

Reply:
I see `.cause` exists in the Couchbase++ library, but I don't see any examples of `->cause` in the PHP wrapper, so unsure how to proceed here.

Reply:
This looks to be an API bug then